src/lib/http.coffee

```diff
@@ -21,12 +23,83 @@ module.exports = (dnschain) ->
             @log = gNewLogger 'HTTP'
             @log.debug "Loading HTTPServer..."
             @rateLimiting = gConf.get 'rateLimiting:http'
+            app = express()
+
+            # Openname spec defined here:
+            # - https://github.com/okTurtles/openname-specifications/blob/resolvers/resolvers.md
+            # - https://github.com/openname/openname-specifications/blob/master/resolvers.md
+
+            opennameRoute = express.Router()
+
+            # Resolver specific API
+            opennameRoute.get /\/(?:resolver|dnschain)\/([^\/\.]+)(?:\.([a-z]+))?/, (req, res) =>
+                @log.debug gLineInfo("resolver API called"), {params: req.params}
+                [resource, format] = req.params
+                if resource == "fingerprint"
+                    if !format or format is 'json'
+                        res.json {fingerprint: @dnschain.encryptedserver.getFingerprint()}
+                    else
+                        @sendErr req, res, 400, "Unsupported format: #{format}"
+                else
+                    @sendErr req, res, 400, "Bad resource: #{resource}"
+
+            # Datastore API
+            opennameRoute.route(/// ^
+                \/(\w+)             # the datastore name
+                \/(\w+)             # the corresponding resource
+                (?:\/([^\/\.]+))?   # optional property (or action on resource)
+                (?:\/([^\/\.]+))?   # optional action on property
+                (?:\.([a-z]+))?     # optional response format
+                $ ///
+            ).get (req, res) =>
+                @log.debug gLineInfo("get v1"), {params: req.params, queryArgs: req.query}
+                @callback  req, res, [_.values(req.params)..., req.query]
+
+            opennameRoute.use (req, res) =>
+                @sendErr req, res, 400, "Bad v1 request"
+
+            app.use "/v1", opennameRoute
+            app.get "*", (req, res) => # Old, deprecated API usage.
```

Comment:
This comment should be changed, as the concept of a metaTLD could still be useful in the future.

Maybe make it: `# metaTLD based API`